CMSgov/easi-app
# EASI 3510/graph schema upgrade

# EASI-3510


## Changes and Description

- upgrade `gqlgen` to v0.17.40
- changes that led to this not being a simple version bump: https://github.com/99designs/gqlgen/pull/2830
- in places where we used `map[string]interface{}` as the `@goModel`, the scalars were coming through as strings instead of their declared scalar types, such as `UUID`, for example. This was fixed in `gqlgen`, but our resolvers were designed with this pattern (always having strings) in mind, so some functionality was updated in this PR to account for the types no longer being strings. Inputs below are where this is relevant
  - `input TRBRequestChanges`
  - `input UpdateTRBRequestFormInput`
  - `input UpdateTRBAdviceLetterInput`
  - `input UpdateTRBAdviceLetterRecommendationInput`



## How to test this change
- pull the branch, confirm all tests pass
- run through any UI functionality that uses the above four inputs, such as:
  - creating _and_ submitting a TRB Request (submission will not work if this change does not work)
  - _updating_ an advice letter
  - _updating_ an advice letter recommendation
  - adding subject areas to a TRB request as you are creating it



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
